packages/contracts/src/hardhat-deploy-ethers.ts

```diff
@@ -67,6 +85,17 @@ export const deployAndRegister = async ({
 
   await hre.ethers.provider.waitForTransaction(result.transactionHash)
 
+  if (postDeployAction) {
+    const signer = hre.ethers.provider.getSigner(deployer)
+    let abi = result.abi
+    if (iface !== undefined) {
+      const factory = await hre.ethers.getContractFactory(iface)
+      abi = factory.interface
+    }
+    const instance = new Contract(result.address, abi, signer)
+    await postDeployAction(instance)
+  }
```

Comment:
Check if `hardhat-ethers` has post-deployment actions already taken into account.

Reply:
It doesn't